lib/generators/generators.py

```diff
@@ -31,11 +32,48 @@ class GeneratorError(Exception):
     """Base class for generator exceptions"""
 
 
+class GeneratorContext(dict):
+    """Context for a generator"""
+
+    def validate(self, valid_attributes: Mapping[str, Tuple[Type, Any]]):
+        """Validate the context against a set of valid attributes.
+        This method ensures that required attributes are present and
+        that their types match the expected types. If an attribute is missing,
+        it will be set to a default value if provided.
+
+        Args:
+            valid_attributes (Mapping[str, Tuple[Type, Any]]): a mapping of attribute names to their expected types and default values
+
+        Raises:
+            GeneratorError: when a required attribute is missing or has an invalid type
+
+        Example:
+            context.validate({
+                "user": str,
+                "group": str,
+                "permissions": list,
+            })
+        """
+
+        for attr, (attr_type, attr_default) in valid_attributes.items():
+            if attr not in self:
+                if attr_default is not None:
+                    self[attr] = attr_default
+                else:
+                    raise GeneratorError(f"Missing required context attribute: {attr}")
+            if not isinstance(self[attr], attr_type):
+                raise GeneratorError(
+                    f"Invalid type for attribute '{attr}': "
+                    f"expected '{attr_type.__name__}', got '{type(self[attr]).__name__}'"
+                )
+
+
 class Generator(ABC, Generic[T]):
     """Base class for generators"""
 
-    def __init__(self, context: Optional[Mapping] = None):
-        self.context = context
+    def __init__(self, context: Optional[Mapping] = None, instance_id: Optional[str] = None):
+        self.context = GeneratorContext(context)
+        self.instance_id = instance_id or hash_nc(f"{time.time()}", 8)
```

Comment:
This instance_id is different and unrelated to the same name used in the _load_generator dictionary?